cmd/cdi-controller/leaderelection.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"os"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/util"
+)
+
+const (
+	configMapName = "cdi-controller-leader-election-helper"
+	componentName = "cdi-controller"
+)
+
+func startLeaderElection(config *rest.Config, leaderFunc func()) error {
+	client := kubernetes.NewForConfigOrDie(config)
+	namespace := util.GetNamespace()
+
+	// crate manually so it has CDI component label
+	err := createConfigMap(client, namespace, configMapName)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
```

Comment:
* Can we simplify to a single if statement for now? I.e. 'if err != nil && ...'
* Might worth it to wrap the error with a specific description.